### PR TITLE
Fix inheritance logic when intermediate class is present

### LINKED ISSUE #4: bug: Inheritance breaks when intermediate class doesn't declare member
### Description of the bug
During the inheriting of docstrings for a member, a direct parent class that doesn't have the member (e.g. `meth`) prevents a correct docstring inheritance from `A.meth -> B.meth`.

### To Reproduce

Run inside this module's environment:

```
from griffe import Extensions, temporary_visited_package

from griffe_inherited_docstrings.extension import InheritDocstringsExtension

code = """
# Base docstrings.
class A:
    def meth(self):
        '''A.'''

class AB(A):
    # This shouldn't break the inherting of docstrings.
    pass

class B(AB):
    def meth(self):
        ...

"""
with temporary_visited_package(
        "package",
        modules={"__init__.py": code},
        extensions=Extensions(InheritDocstringsExtension(merge=True)),
    ) as package:
        assert package["B.meth"].docstring # already fails here
        assert package["B.meth"].docstring.value == package["A.meth"].docstring.value
```

</details>

### Expected behavior

I expect the inheritance to work.

### Environment information
- __Python__: cpython 3.9.18 (/Users/XXXX/griffe-inherited-docstrings/.venv/bin/python)
- __Environment variables__:
- __Installed packages__:
  - `griffe-inherited-docstrings` v1.1.0+d20241029

### PR DESCRIPTION
**Pull request by thomasmarwitz**
This fixes #4

### CODE REVIEW

**Review by pawamoy — CHANGES_REQUESTED** [review]
Could you add a new test rather than modify an existing one?

**Comment by pawamoy** [comment]
I tried this locally, and unfortunately the test passes even without the fix.

File: tests/test_extension.py [hunk 1]
```diff
@@ -87,3 +87,32 @@ def meth(self):
...
+                class Parent:
+                    def method(self):
+                        '''Parent.'''
+
+                class Intermediate(Parent):
+                    # This shouldn't break the inherting of docstrings.
+                    # See https://github.com/mkdocstrings/griffe-inherited-docstrings/issues/4
+                    ...
+
+                class Child(Parent):
```
Comment (pawamoy) [inline tests/test_extension.py:111]:
```suggestion
                class Child(Intermediate):
```

Comment (thomasmarwitz) [reply tests/test_extension.py:111]: Ah, thanks for pointing that out >.<

**Review by pawamoy — APPROVED** [review]
OK LGTM, thank you for following up :smile:

**Comment by thomasmarwitz** [comment]
Thanks for double-checking everything :)

**Comment by pawamoy** [comment]
Released in v1.1.1.

**Pull request merged**